src/o365/graph/commands/reports/reports-teams-device-usage-user-detail.ts

```diff
@@ -0,0 +1,153 @@
+    return new Promise<void>((resolve: (data: any) => void, reject: (error: any) => void): void => {
+      auth
+        .ensureAccessToken(auth.service.resource, cmd, this.debug)
+        .then((): Promise<any> => {
+          const requestOptions: any = {
+            url: url,
+            headers: {
+              authorization: `Bearer ${auth.service.accessToken}`
+            }
+          };
+
+          return request.get(requestOptions);
+        })
+        .then((res: any): void => {
+          resolve(res);
+        }, (err: any): void => {
+          reject(err);
+        });
+    });
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-p, --period [period]',
+        description: 'Specify the length of time over which the report is aggregated. The supported values are D7|D30|D90|D180',
+        autocomplete: ['D7', 'D30', 'D90', 'D180']
+      },
+      {
+        option: '-d, --date [date]',
+        description: 'Specify the date for which you would like to view the users who performed any activity. The supported date format is YYYY-MM-DD'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+
+    return (args: CommandArgs): boolean | string => {
+
+      if (args.options.period && args.options.date) {
+        return 'You can\'t use period and date parameters together.';
+      }
+
+      if (args.options.period) {
+        const period: string = args.options.period.toUpperCase();
+        if (period !== 'D7' && period !== 'D30' && period !== 'D90' && period !== 'D180') {
+          return `${period} is not a valid period type. The supported values are D7|D30|D90|D180`;
+        }
+      }
+
+      const parsedDate = Date.parse(args.options.date as string);
+
+      if (args.options.date && !(!parsedDate) !== true) {
+        return `Provide the date in YYYY-MM-DD format`;
+      }
+
+      if (new Date(parsedDate) <= maxDate) {
+        return `This report is only available for the past 30 days, date value should be a date from that range.`;
+      }
+
+      return true;
+    };
+  }
+
+  public commandHelp(args: {}, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(this.name).helpInformation());
+    log(
+      `  ${chalk.yellow('Important:')} before using this command, log in to the Microsoft Graph
+    using the ${chalk.blue(commands.LOGIN)} command.
+        
+  Remarks:
+
+    To get details about Microsoft Teams device usage by user, you have to first log in to
+    the Microsoft Graph using the ${chalk.blue(commands.LOGIN)} command,
+    eg. ${chalk.grey(`${config.delimiter} ${commands.LOGIN}`)}.
+    
+    Reports.Read.All permission is required to call this API.
+
+  Examples:
+
+  Gets detail about Microsoft Teams device usage by user for the length of time over which 
```

Comment:
Let's simplify this part `for the length of time over which the report is aggregated` by saying `for the last 7 days`

Reply:
Fixed and ready for review.